src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs

```diff
@@ -1522,6 +1522,147 @@ void M()
             }
         }
 
+        [WorkItem(31304, "https://github.com/dotnet/roslyn/issues/31304")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TestCompletionDoesNotUseForeachVariableName()
+        {
+            var markup = @"
+class ClassA
```

Comment:
What is the behavior in VB?

Reply:
I don't get any VB name suggestions at all in the editor and as far as I can tell there are no tests for name suggestions in VB either.  My guess is because the type declaration happens after the variable name is chosen e.g. ``Dim classB As ClassB``.  Similarly in C# there are not any name suggestions when using ``var``.  

The DeclarationNameCompletionProvider seems to be only be used by the CSharpCompletionService and is in a CSharp package.

Reply:
Thanks for checking!